pylint/checkers/variables.py

```diff
@@ -2940,7 +2930,19 @@ def visit_const(self, node: nodes.Const) -> None:
             if origin is not None and utils.is_typing_literal(origin):
                 return
 
-        self._type_annotation_names.append(node.value)
+        if node.value.isidentifier():
```

Comment:
Could you explain why you moved this?

Reply:
I moved the is_typing_literal part up to stop checking for names if the string node is an arg to `Literal`. As for the `isidentifier` check, it's probably not needed anymore. I'll push some more changes.